File: Examples/runtimes/java/DynamoDbEncryption/src/test/java/software/amazon/cryptography/examples/keyring/TestRawRsaKeyringExample.java
package software.amazon.cryptography.examples.keyring;

import java.util.UUID;
import org.testng.annotations.Test;
import software.amazon.cryptography.examples.TestUtils;

public class TestRawRsaKeyringExample {

  @Test
  public void TestRawRsaKeyringExample() {
    final String partitionKeyValue = "RawRsaKeyringExample" + UUID.randomUUID();
    // You may provide your own RSA key pair in the files located at
    //  - EXAMPLE_RSA_PRIVATE_KEY_FILENAME
    //  - EXAMPLE_RSA_PUBLIC_KEY_FILENAME
    // If these files are not present, this will generate a pair for you
    if (RawRsaKeyringExample.shouldGenerateNewRsaKeyPair()) {
      RawRsaKeyringExample.generateRsaKeyPair();
    }

    RawRsaKeyringExample.RawRsaKeyringGetItemPutItem(
      TestUtils.TEST_DDB_TABLE_NAME,
      partitionKeyValue,
      "0"
    );
    
    TestUtils.cleanUpDDBItem(
      TestUtils.TEST_DDB_TABLE_NAME,
      "partition_key",
      "sort_key",
      partitionKeyValue,
      "0"
    );
  }
}

File: Examples/runtimes/python/DynamoDBEncryption/test/keyring/test_raw_rsa_keyring_example.py
# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0
"""Test for the Raw RSA keyring example."""
import pytest

from ...src.keyring.raw_rsa_keyring_example import (
    generate_rsa_key_pair,
    raw_rsa_keyring_example,
    should_generate_new_rsa_key_pair,
)
from ..test_utils import TEST_DDB_TABLE_NAME

pytestmark = [pytest.mark.examples]


def test_raw_rsa_keyring_example():
    """Test the Raw RSA keyring example."""
    # You may provide your own RSA key pair in the files located at
    #  - EXAMPLE_RSA_PRIVATE_KEY_FILENAME
    #  - EXAMPLE_RSA_PUBLIC_KEY_FILENAME
    # If these files are not present, this will generate a pair for you
    if should_generate_new_rsa_key_pair():
        generate_rsa_key_pair()

    raw_rsa_keyring_example(TEST_DDB_TABLE_NAME)
